Algorithm.Framework/Selection/EmaCrossUniverseSelectionModel.cs

```diff
@@ -0,0 +1,102 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Indicators;
+using QuantConnect.Securities;
+
+namespace QuantConnect.Algorithm.Framework.Selection
+{
+    /// <summary>
+    /// Provides an implementation of <see cref="FundamentalUniverseSelectionModel"/> that subscribes
+    /// to symbols with the larger delta by percentage between the two exponential moving average
+    /// </summary>
+    public class EmaCrossUniverseSelectionModel : FundamentalUniverseSelectionModel
+    {
+        private const decimal _tolerance = 0.01m;
+        private readonly int _fastPeriod;
+        private readonly int _slowPeriod;
+        private readonly int _universeCount;
+        
+        // holds our coarse fundamental indicators by symbol
+        private readonly ConcurrentDictionary<Symbol, SelectionData> _averages;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmaCrossUniverseSelectionModel"/> class
+        /// </summary>
+        /// <param name="fastPeriod">Fast EMA period</param>
+        /// <param name="slowPeriod">Slow EMA period</param>
+        /// <param name="universeCount">Maximum number of members of this universe selection</param>
+        /// <param name="universeSettings">The settings used when adding symbols to the algorithm, specify null to use algorthm.UniverseSettings</param>
+        /// <param name="securityInitializer">Optional security initializer invoked when creating new securities, specify null to use algorithm.SecurityInitializer</param>
+        public EmaCrossUniverseSelectionModel(
+            int fastPeriod = 100,
+            int slowPeriod = 300,
+            int universeCount = 10000,
```

Comment:
default of 10k securities! whoa!

Reply:
I had `int.MaxValue`, but I will bring it down to 100 and let the users increase it instead of decrease it.